### PR TITLE
Add team configuration modal for users and api users

### PR DESCRIPTION
**Pull request by TheWalkingLeek**

### CODE REVIEW

File: frontend/app/templates/components/team-member-configure.hbs [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+          </li>
+        </ul>
+        <div class="tab-content">
+          <div class="tab-pane active" id="members" role="tabpanel" aria-labelledby="members-tab">
+            <div class="columns members">
+              <div class="row justify-content-end">
+                <div class="col-sm-4 float-right">
+                  <PowerSelectTypeahead @renderInPlace="true" @options={{this.candidates}} @search={{this.search}} @onChange={{this.addMember}} as |candidate|>
+                    {{candidate.label}}
+                  </PowerSelectTypeahead>
```
Comment (lucashabersaat) [inline frontend/app/templates/components/team-member-configure.hbs:26]: Not yet properly styled. Not sure how to add this:
https://github.com/cibernox/ember-power-select-typeahead#styles

**Comment by lucashabersaat** [comment]
Currently, assets are not like we want them. When ember builds, the assets are moved into _public/ember/assets_. And ideally, they are in _frontend/public/assets/_ . But to be able to correctly link them in development environment and after building in production, (so two different directories) we need some solution. 
Currently, images only load when ember has built.

File: frontend/app/adapters/api-user.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import ApplicationAdapter from "./application";
+import { pluralize } from "ember-inflector";
+
+export default ApplicationAdapter.extend({
+  pathForType(modelName) {
+    let snake_cased = modelName.replace("-", "_");
```
Comment (psunix) [inline frontend/app/adapters/api-user.js:6]: snakeCased ?
I know, almost silly :joy: 
but we always use camelCase for JS

File: frontend/app/components/team-member-configure.js [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+        });
+      }
+    );
+  }
+
+  abort() {
+    this.router.transitionTo("index");
+  }
+
+  handleSubmitSuccessful() {
```
Comment (psunix) [inline frontend/app/components/team-member-configure.js:45]: handleSubmitSuccess ?

File: frontend/app/components/team-member-configure.js [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+      this.enableApiUser(apiUser).then(() => (apiUser.enabled = true));
+    }
+  }
+
+  enableApiUser(apiUser) {
+    /* eslint-disable no-undef  */
+    return fetch(`/api/teams/${this.args.teamId}/api_users`, {
+      method: "post",
+      headers: {
+        "Content-type": "application/x-www-form-urlencoded; charset=UTF-8",
```
Comment (psunix) [inline frontend/app/components/team-member-configure.js:91]: this looks like it should be somewhere central ?
but maybe this is going to change with the json api implementation ?

File: frontend/app/serializers/api-user.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
+import ApplicationSerializer from "./application";
+import { underscore } from "@ember/string";
+
+export default ApplicationSerializer.extend({
+  modelNameFromPayloadKey(payloadKey) {
+    if (payloadKey === "user/apis" || payloadKey === "team/api_users") {
```
Comment (psunix) [inline frontend/app/serializers/api-user.js:6]: nice !

File: frontend/tests/integration/components/team-member-configure-test.js [hunk 6]
```diff
@@ -0,0 +1,35 @@
+import { module, test } from "qunit";
```
Comment (psunix) [inline frontend/tests/integration/components/team-member-configure-test.js:1]: awesome, we got also those nice frontend tests !

**Comment by psunix** [comment]
don't forget to rebase to master !
it looks like there's some changes that are not yet on this branch

File: frontend/app/components/team-member-configure.js (hunk 4 above)
Comment (lucashabersaat) [reply frontend/app/components/team-member-configure.js:91]: Probably yes, will check with @TheWalkingLeek and move to json story.

**Pull request merged**